src/db/oracledb/jobs-dao.js

```diff
@@ -56,4 +120,269 @@ const getJobByJobId = async (internalId, jobId) => {
+ * @returns {string} Query result, null if success
+ */
+const graduateJob = async (connection, osuId, body, update) => {
+  const binds = standardBinds(osuId, body, graduateBinds);
+  const queryName = update === 'update' ? 'graduateUpdateJob' : 'graduateCreateJob';
+
+  const { outBinds: { result } } = await connection.execute(
+    contrib[queryName](binds),
+    binds,
+  );
+  return result;
+};
+
+/**
+ * Determines if changeReasonCode is valid
+ *
+ * @param {object} connection oracledb connection
+ * @param {string} changeReasonCode Change reason code from request body
+ * @returns {boolean} True if changeReasonCode is valid
+ */
+const isValidChangeReasonCode = async (connection, changeReasonCode) => {
+  const { rows } = await connection.execute(
+    contrib.validateChangeReasonCode(),
+    { changeReasonCode },
+  );
+  return rows[0].count > 0;
+};
+
+/**
+ * Determines which Epaf to execute based on fields in body
+ *
+ * @param {string} update 'create' to create a new record, 'update' to update an existing one
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @param {string} internalId Internal ID of a person
+ * @returns {object} raw job record
+ */
+const createOrUpdateJob = async (update, osuId, body) => {
+  const connection = await getConnection('banner');
+  try {
+    let result;
+    const { studentEmployeeInd, changeReason: { code: changeReasonCode } } = body;
+    const employmentType = studentEmployeeInd ? 'student' : 'graduate';
+
+    if (employmentType === 'student' && !/^C5[0-2]/.test(body.positionNumber)) {
+      return new Error('Position number for students must start with C50, C51, or C52');
+    }
+
+    if (_.includes(['TERME', 'TERMJ'], changeReasonCode)) {
+      result = await terminateJob(connection, osuId, body);
+    } else {
+      if (!await isValidChangeReasonCode(connection, changeReasonCode)) {
+        return new Error(`Invalid change reason code ${changeReasonCode}`);
+      }
+
+      if (update === 'update') {
+        if (changeReasonCode === 'NONE') {
+          result = await updateLaborChangeJob(connection, osuId, body);
+        } else if (changeReasonCode === 'BREAP') {
+          if (employmentType === 'student') {
+            result = await studentUpdateJob(connection, osuId, body);
+          } else if (employmentType === 'graduate') {
+            result = await graduateJob(connection, osuId, body, update);
+          }
+        } else {
+          await updateJob(connection, osuId, body);
+        }
+      } else if (employmentType === 'student') {
+        result = await studentCreateJob(connection, osuId, body);
+      } else if (employmentType === 'graduate') {
+        result = await graduateJob(connection, osuId, body, update);
+      }
+    }
+
+    // success when result is null
+    if (!result) {
+      // connection.commit();
+      return result;
```

Comment:
Based on the JSDoc, the returned value should be an object of raw job record, but it looks like we're only returning error or null (if succeed) here?

Reply:
Good catch. forgot to update the jsdoc